### PR TITLE
add txpool namespace support

### PR DESCRIPTION
**Pull request by ksato9700**
This is a PR for the issue #381. Please review and merge it if you like.

### CODE REVIEW

File: src/types/txpool.rs [hunk 1]
```diff
@@ -0,0 +1,204 @@
+use std::collections::HashMap;
+use crate::types::{Transaction};
+use serde::{Deserialize, Serialize};
+use crate::types::U64;
+
+/// Transaction Pool Content Info
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
+pub struct TxpoolContentInfo {
+    /// pending tx
+    pub pending: HashMap<String, HashMap<String, Transaction>>,
```
Comment (tomusdrw) [inline src/types/txpool.rs:10]: Perhaps better to use `BTreeMap` here to have the keys deterministically ordered? With `HashMap` the results would be a bit random. Also we should be using more strict types than `String`. I believe it's transaction hash and the nonce, right?
```suggestion
    pub pending: BTreeMap<H256, BTreeMap<U256, Transaction>>,
```

File: src/types/txpool.rs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    /// queued tx
+    pub queued: HashMap<String, HashMap<String, Transaction>>,
```
Comment (tomusdrw) [inline src/types/txpool.rs:12]:
```suggestion
    pub queued: BTreeMap<H256, BTreeMap<U256, Transaction>>,
```

File: src/types/txpool.rs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+/// Transaction Pool Inspect Info
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
+pub struct TxpoolInspectInfo {
+    /// pending tx
+    pub pending: HashMap<String, HashMap<String, String>>,
```
Comment (tomusdrw) [inline src/types/txpool.rs:19]:
```suggestion
    pub pending: BTreeMap<H256, BTreeMap<U256, String>>,
```

File: src/types/txpool.rs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    /// queued tx
+    pub queued: HashMap<String, HashMap<String, String>>,
```
Comment (tomusdrw) [inline src/types/txpool.rs:21]:
```suggestion
    pub queued: BTreeMap<H256, BTreeMap<U256, String>>,
```

File: src/types/txpool.rs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+/// Transaction Pool Content Info
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
+pub struct TxpoolContentInfo {
+    /// pending tx
+    pub pending: HashMap<String, HashMap<String, Transaction>>,
+    /// queued tx
+    pub queued: HashMap<String, HashMap<String, Transaction>>,
+}
+
+/// Transaction Pool Inspect Info
```
Comment (tomusdrw) [inline src/types/txpool.rs:15]: Can we expand on the docs or at least point to documentation of these structs? I'd be more interested in the `String` format of the transaction inspection info - i.e. what things can be found there and can we actually parse it?

File: src/types/txpool.rs [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+        let _: TxpoolInspectInfo = serde_json::from_str(txpool_inspect_str).unwrap();
+    }
+
+    #[test]
+    fn should_deserialize_txpool_status() {
+        let txpool_status_str = r#"{
+            "pending":"0x23",
+            "queued":"0x20"
+        }"#;
+        let _: TxpoolStatus = serde_json::from_str(txpool_status_str).unwrap();
```
Comment (tomusdrw) [inline src/types/txpool.rs:202]: Re-serialization test would be cool, after switching to `BTreeMap` you will get consistent results.

**Review by tomusdrw — COMMENTED** [review]
Looks really good for a start, thanks! I'd like to see a bit more strict typing and some documentation pointers.

**Comment by ksato9700** [comment]
@tomusdrw Thank you for the review comments. I made changes as follows.
* use `BTreeMap` instead of `HashMap`
* use `Address` for the type of outer key
* add some comments with link to a spec document
* add re-serialization tests

Note that I keep using `String` for the inner key type. As you said, it is nonce, but strangely enough, decimal number format is used here instead of hex.

**Review by tomusdrw — APPROVED** [review]
looks awesome, thanks!

**Pull request merged**